src/main.cpp

```diff
@@ -0,0 +1,135 @@
+    degrees y_angle;
+    degrees x_angle;
+};
+
+/**
+    This function will call the treed_wrapper node to generate scans.
+    Every scan will be stored as an scanData object which consist of angles of the board and the point cloud. 
+    @param accuracy This will set the precision on the scan, in other words, how many times to scan.
+    @param scans The vector to store all scanData objects in.
+    @return true if success, false otherwise.
+*/
+bool
+generate_scans(uint16 accuracy, std::vector<scanData> &scans) 
+{
+    // Constant number for the degrees of the rotation board.
+    degrees y_min = 0;
+    degrees y_max = 359;
+    degrees x_min = -20;
+    degrees x_max = 90;
+    
+    // Define how many degrees to rotate every time.
+    degrees y_rotate = 100 / accuracy;
+    degrees x_rotate = 90 / accuracy;
+
+    // Define how many times to scan in both y and x directions.
+    double times_to_scan_x = std::floor((x_max - x_min) / x_rotate);
+    double times_to_scan_y = std::floor((y_max - y_min) / y_rotate);
+    
+    // Define the angles for y and x axis.
+    degrees y_angle;
+    degrees x_angle = x_min;
+    
+    // Define the client that will be used to call the wrapper_scan service. 
+    ros::NodeHandle node_handle;
+    ros::ServiceClient client = node_handle.serviceClient<scanService>("wrapper_scan");
+    
+    // For every angle in x, rotate the board on every angle on the y axis.    
+    for (int i = 0; i < times_to_scan_x; ++i) {
+        y_angle = y_min;
+        for (int j = 0; j < times_to_scan_y; ++j) {
+            y_angle = y_angle + y_rotate;
+            
+            // Setup the scan service and request the service to set the angles.
+            scanService srv;
+            srv.request.y_angle = y_angle;
+            srv.request.x_angle = x_angle;
+            
+            // Call the scanService (wrapper), it will return true if service call succeeded, it will return false if the call not succeed.
+            // It will also check if the exit_code is valid (return 0).
+            if (client.call(srv) && !srv.response.exit_code) {
+                scanData scan_data;
+                scan_data.y_angle = y_angle;
+                scan_data.x_angle = x_angle;
+                scan_data.point_cloud_message = srv.response.point_cloud;
+                scans.push_back(scan_data);
+            } else {
+                return false;
+            }
+        }
+        x_angle = x_angle + x_rotate;
+    }
+    return true;
+}
+
+/**
+    This function will get the gather point clouds that will be stored in a vector. 
+    @param req The service request that has been made.
+    @param resp The response, vector which contains the scanned point clouds.
+    @return true when the process to gathering and regiser point clouds is done.
+*/
+bool 
+get_point_cloud(getPointCloud::Request &req, getPointCloud::Response &resp)
+{
+    // Check if the accuracy is valid.
+    if (req.accuracy <= 0) {
```

Comment:
Vi har väl sag att accuracy ska vara mellan 0-10 så vi kan väl kolla hela den grejen så man inte skickar in 100 och får vänta i år och dar.